File: back/back/Models/Logistique.cs
using LiteDB;

namespace back.Models;

public sealed class Logistique
{
     [BsonId]
     public int Id { get; set; }

     [BsonRef]
     public LogistiqueType LogistiqueType { get; set; } = null!;

     [BsonRef]
     public TypeStockageLogistique TypeStockage { get; set; } = null!;
     public string Nom { get; set; } = null!;
     public int Prix { get; set; }
     public int Stock { get; set; }
     public int NbDetruit { get; set; }
     public int TailleUnitaireInventaire { get; set; }
     public bool IgnoreTypeStockage { get; set; }
     public bool EstKit { get; set; }
}

File: front/src/models/Logistique.ts
export type Logistique =
{
    kind: "Logistique",
    id: number,
    nom: string,
    description: string,
    prix: number,
    nbDetruit: number,
    listeStockageVaisseau: LogistiqueStockageVaisseau[],
    tailleUnitaireInventaire: number,
    estKit: boolean,
    ignoreTypeStockage: boolean,
    type: TypeLogistique,
    typeStockage: TypeStockageLogistique
}

export type LogistiqueRequete = Omit<Logistique, "id" | "listeStockageVaisseau">;

export type LogistiqueStockageVaisseau =
{
    nomVaisseau: string,
    nomStockage: string,
    quantite: number
}

export type TypeLogistique = 
{
    id: number,
    nom: string
}

export type TypeStockageLogistique = 
{
    id: number,
    nom: string
}